Source/WebCore/platform/graphics/avfoundation/AudioVideoRendererAVFObjC.mm

```diff
@@ -157,11 +157,37 @@ @interface AVSampleBufferDisplayLayer (Staging_100128644)
         return;
 
     switch (*type) {
-    case TrackType::Video:
+    case TrackType::Video: {
+        PlatformSample platformSample = sample->platformSample();
+        CMFormatDescriptionRef formatDescription = PAL::CMSampleBufferGetFormatDescription(platformSample.sample.cmSampleBuffer);
+        ASSERT(formatDescription);
+        if (!formatDescription) {
+            ERROR_LOG(LOGIDENTIFIER, "Received sample with a null formatDescription. Bailing.");
+            return;
+        }
+        auto mediaType = PAL::CMFormatDescriptionGetMediaType(formatDescription);
+        ASSERT(mediaType == kCMMediaType_Video);
+        if (mediaType != kCMMediaType_Video) {
+            ERROR_LOG(LOGIDENTIFIER, "Expected sample of type: '", FourCC(kCMMediaType_Video), "', got: '", FourCC(mediaType), "'. Bailing.");
+            return;
+        }
+
+        if (m_sizeChangedCallback) {
+            FloatSize formatSize = FloatSize(PAL::CMVideoFormatDescriptionGetPresentationDimensions(formatDescription, true, true));
```

Comment:
Nit: now that we have it, we should probably use `FormatDescriptionUtilities` here.